drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -525,16 +519,16 @@ static int i2c_mspm0g3xxx_target_unregister(const struct device *dev,
 	} else {
 		LOG_ERR("Failed to unregister device as none was found at addr=%u",
 			target_config->address);
-		k_sem_give(&data->i2c_busy_sem);
+
 		return -ENODEV;
 	}
 
 	/* Check if no targets are registered - we'll then re-enable controller mode */
 	if (data->target_config_primary != NULL || data->target_config_alternate != NULL) {
-		k_sem_give(&data->i2c_busy_sem);
 		return 0;
 	}
 
+	k_sem_take(&data->i2c_busy_sem, K_FOREVER);
```

Comment:
Just so I follow 100%, this will never be called in an isr context? If that's not guaranteed, the semaphore needs to either be replaced, or always taken with `K_NO_WAIT` (api states this)

Reply:
Yeah, I was thinking if there is no guarantee, we should use the workqueue which is already there. Super. I will take another look later today. The diff needs some closer inspection this time around ;)

Reply:
Exactly! This part of the driver always crashed when enabled assertions so I wanted to fix that for a while :laughing: it's from the "upstream TI driver". But they haven't added the msgq (yet at least), so I think it's fine that we deviate as long as we're utilizing the mechanisms that are already deviating :)